controls/tests.py

```diff
@@ -229,29 +228,45 @@ def test_component_download_oscal_json(self):
         # downloaded file, so let's make sure it doesn't exist before we
         # download
         # definite race condition possibility
-        
-        if os.path.isfile(self.json_download.name):
+
+        if self.json_download.is_file():
+            self.json_download.unlink()
+        elif os.path.isfile(self.json_download.name):
             os.remove(self.json_download.name)
         self.click_element("a#oscal_download_json_link")
         var_sleep(2)            # need to wait for download, alas
         # assert download exists!
-        self.assertTrue(os.path.isfile(self.json_download.name))
-        # assert that it is valid JSON by trying to load it
-        with open(self.json_download.name, 'r') as f:
+        try:
+            self.assertTrue(self.json_download.is_file())
+            filetoopen = self.json_download
+        except:
+            self.assertTrue(os.path.isfile(self.json_download.name))
+            # assert that it is valid JSON by trying to load it
+            filetoopen = self.json_download.name
+        with open(filetoopen, 'r') as f:
             json_data = json.load(f)
             self.assertIsNotNone(json_data)
-        os.remove(self.json_download.name)
+
+        if self.json_download.is_file():
+            self.json_download.unlink()
+        elif os.path.isfile(self.json_download.name):
+            os.remove(self.json_download.name)
```

Comment:
Is it possible to determine ahead of time which OS we're using, rather than interspersing the OS-specific checks throughout the test? I found it a little confusing to follow if these checks are done for specific test reasons, or to accommodate OS-specific functionality, and thought it might be easier to separate those two logic trees, architecturally.

Reply:
Unfortunately no, normally I would just check the platform but within a selenium test it used the dos file path system instead of linux(WSL).